torch/csrc/jit/tensorexpr/fwd_decls.h

```diff
@@ -1,26 +1,27 @@
 #pragma once
 #include <c10/core/ScalarType.h>
+#include <memory>
 
 namespace torch {
 namespace jit {
 namespace tensorexpr {
 
 template <typename Node>
-using NodePtr = Node*;
+using NodePtr = std::shared_ptr<Node>;
```

Comment:
Is it possible to keep this change, i.e., switching to `shared_ptr` alone separate from all the other changes in the diff? If there is any problem and you have to revert, you will only revert a much smaller diff that way. 

Just a suggestion...

Reply:
Good point. Initially I did that, but then CI detected some failures in the code I wasn't testing locally so some more prep code changes creeped in into this PR. I split that into #63778. This PR can't be reduced to a single line change, but overall now it should be reviewable and mostly contain the relevant to shared_ptr switch changes.